en/docs/learn/configuring-totp.md

```diff
@@ -0,0 +1,398 @@
+            Authenticators** of Step 1 to add the `basic` authentication
+            as the first step.  
+            Adding basic authentication as a first step ensures that the
+            first step of authentication will be done using the user's
+            credentials that are configured with the WSO2 Identity
+            Server
+
+    3.  Creating the second authentication step:
+
+        1.  Click **Add Authentication Step**.
+
+        2.  Click **Add Authenticator** that is under **Local
+            Authenticators** of Step 2 to add the `totp` from the drop
+            down as the second step.  
+            ![creating-the-second-authentication-step-totp](../assets/img/tutorials/creating-the-second-authentication-step-totp.png)
+            
+Now `totp' is a second step that adds another layer of authentication
+and security during the authentication for this application.
+              
+## Configuring the user profile
+
+Now you have to register for the TOTP authentication from your user
+profile.
+
+1.  Login to the
+    [end user dashboard](../../learn/using-the-end-user-dashboard) and
+    go to **My Profile** by clicking **View details** .  
+    ![my-profile-gadget](../assets/img/tutorials/my-profile-gadget.png)
+4.  Update your email address (this email address is used to send the
+    token).
+3.  To enable TOTP, select the **Enable TOTP** checkbox.  
+    ![scan-totp-qr](../assets/img/tutorials/scan-qr.png)
+4.  You can use the [Google Authenticator](https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en)
+    Application to generate the one-time passwords (tokens), click on
+    **Scan QR Code** to scan the QR-Code using the Google Authenticator
+    mobile app.
+
+## Testing the sample
+
+1.  To test the sample, go to the following URL:
+    `            http://<TOMCAT_HOST>:<TOMCAT_PORT>/travelocity.com/index.jsp.           `
+    For example, this looks like
+    [http://localhost:8080/travelocity.com.  
+    ](http://localhost:8080/travelocity.com)
+
+2.  Click the link to log in with SAML from the WSO2 Identity Server.  
+3.  The basic authentication page is visible. Use your username and
+    password to log in.  
+4.  If the TOTP is not enabled the user's profile and the user is
+    allowed to enable the TOTP in the authentication flow, this page
+    will appear. You can scan either continue or cancel.  
+    ![](../assets/img/tutorials/totp-register.png)
+5.  If you want to enrolthe user, click on the link to show the QR code.
+    Scan the displayed QR code using the mobile application and
+    continue. 
+    
+    ??? Abstract "Click here to view QR-Code display page"
+        ![](../assets/img/tutorials/scan-qr-auth-flow.png)
+        
+6.  You are redirected to the TOTP authentication page. Enter the
+    verification code from your Google Authenticator Mobile Application
+    to authenticate. 
+    ![](../assets/img/tutorials/enter-totp.png)
```

Comment:
Note that you are redirected to the TOTP authenticaion page.

6.   Enter the ....